Mathlib/Algebra/Module/Injective.lean

```diff
@@ -0,0 +1,467 @@
+      (by rw [← eq2]; exact Submodule.sub_mem _ (ExtensionOfMaxAdjoin.fst i x).2 ha)
+  simp only [map_sub, sub_smul, sub_eq_iff_eq_add] at eq3
+  unfold ExtensionOfMaxAdjoin.extensionToFun
+  rw [eq3, ← add_assoc, ← (extensionOfMax i f).toLinearPMap.map_add, AddMemClass.mk_add_mk]
+  congr
+  ext
+  dsimp
+  rw [Subtype.coe_mk, add_sub, ← eq1]
+  exact eq_sub_of_add_eq (ExtensionOfMaxAdjoin.eqn i x).symm
+set_option linter.uppercaseLean3 false in
+#align module.Baer.extension_of_max_adjoin.extension_to_fun_wd Module.Baer.ExtensionOfMaxAdjoin.extensionToFun_wd
+
+set_option synthInstance.etaExperiment true in
+/-- The linear map `M ⊔ ⟨y⟩ ⟶ Q` by `x + r • y ↦ f x + φ r` is an extension of `f`-/
+def extensionOfMaxAdjoin (h : Module.Baer R Q) (y : N) : ExtensionOf i f where
+  domain := supExtensionOfMaxSingleton i f y -- (extensionOfMax i f).domain ⊔ Submodule.span R {y}
+  le := le_trans (extensionOfMax i f).le le_sup_left
+  toFun :=
+    { toFun := ExtensionOfMaxAdjoin.extensionToFun i f h
+      map_add' := fun a b => by
+        have eq1 :
+          ↑a + ↑b =
+            ↑(ExtensionOfMaxAdjoin.fst i a + ExtensionOfMaxAdjoin.fst i b) +
+              (ExtensionOfMaxAdjoin.snd i a + ExtensionOfMaxAdjoin.snd i b) • y := by
+          rw [ExtensionOfMaxAdjoin.eqn, ExtensionOfMaxAdjoin.eqn, add_smul, Submodule.coe_add]
+          ac_rfl
+        rw [ExtensionOfMaxAdjoin.extensionToFun_wd (y := y) i f h (a + b) _ _ eq1,
+          LinearPMap.map_add, map_add]
+        unfold ExtensionOfMaxAdjoin.extensionToFun
+        abel
+      map_smul' := fun r a => by
+        dsimp
+        have eq1 :
+          r • (a:N) =
+            ↑(r • ExtensionOfMaxAdjoin.fst i a) + (r • ExtensionOfMaxAdjoin.snd i a) • y := by
+          rw [ExtensionOfMaxAdjoin.eqn, smul_add, smul_eq_mul, mul_smul]
+          rfl
+        rw [ExtensionOfMaxAdjoin.extensionToFun_wd i f h (r • a) _ _ eq1, LinearMap.map_smul,
+          LinearPMap.map_smul, ← smul_add]
+        congr }
+  is_extension m := by
+    dsimp
+    simp only [LinearPMap.mk_apply, LinearMap.coe_mk]
+    rw [(extensionOfMax i f).is_extension,
+      ExtensionOfMaxAdjoin.extensionToFun_wd i f h _ ⟨i m, _⟩ 0 _, map_zero, add_zero]
+    simp
+set_option linter.uppercaseLean3 false in
+#align module.Baer.extension_of_max_adjoin Module.Baer.extensionOfMaxAdjoin
+
+set_option synthInstance.etaExperiment true in
+theorem extensionOfMax_le (h : Module.Baer R Q) {y : N} :
+    extensionOfMax i f ≤ extensionOfMaxAdjoin i f h y :=
+  ⟨le_sup_left, fun x x' EQ => by
+    symm
+    change ExtensionOfMaxAdjoin.extensionToFun i f h _ = _
+    rw [ExtensionOfMaxAdjoin.extensionToFun_wd i f h x' x 0 (by simp [EQ]), map_zero,
+      add_zero]⟩
+set_option linter.uppercaseLean3 false in
+#align module.Baer.extension_of_max_le Module.Baer.extensionOfMax_le
+
+theorem extensionOfMax_to_submodule_eq_top (h : Module.Baer R Q) :
+    (extensionOfMax i f).domain = ⊤ := by
+  refine' Submodule.eq_top_iff'.mpr fun y => _
+  dsimp
```

Comment:
Similar comment here. Is the fact that we need `dsimp` evidence of some bigger problem somehow?

Reply:
I think it is just the world we live in now.